lib/CL/pocl_llvm_build.cc

```diff
@@ -520,6 +524,14 @@ int pocl_llvm_build_program(cl_program program,
   if (device->llvm_abi != NULL)
     ss << "-target-abi " << device->llvm_abi << " ";
 
+#if defined(__riscv)
+  if (device->llvm_arch != NULL)    {
+    char features[4096] = { '\0' };
+    pocl_get_llvm_cpu_feature(device->llvm_arch, features, 4096);
```

Comment:
This would be better called only once. Also these device-driver-specific flags should be added in the init_build() hook in the CPU driver. Let's try reduce driver-specific codes in this file which should be quite generic (to all LLVM targets).